### PR TITLE
Hosting and deployment updates (Stage 1)

### LINKED ISSUE #4821: CSAT Host ASP.NET Core on Windows with IIS
[URL](https://docs.microsoft.com/en-us/aspnet/core/publishing/iis?tabs=aspnetcore2x)

Plan:
- [ ]  UE pass
- [ ] split off the Troubleshooting into a new standalone topic

### LINKED ISSUE #5055: Document reasoning for web.config location
Per discussion with @spottedmahn: Document the reason why *web.config* was moved from the webroot to the approot ...

> Moving the web.config file location to the root of the application enables publishing of multiple applications using Web Deploy.

Given the length, complexity, and focus of the hosting in IIS topic, this would probably be best in the [Directory structure](https://docs.microsoft.com/aspnet/core/hosting/directory-structure) topic.

It's probably also a good idea given the security risks of the new layout to either state explicitly what the workaround is for relocating the file back into *wwwroot* or at least link out to the GH issue in the *Additional resources* section. https://github.com/aspnet/IISIntegration/issues/164

@Rick-Anderson @scottaddie I'll knock this out on Jan. 2 if that's ok.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #4821
Addresses #4866 
Fixes #5055 
Addresses #5100

[Internal Review Link (Host and deploy landing page)](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/index?branch=pr-en-us-5127&tabs=aspnetcore2x)

* Revises the TOC to move hosting and deployment topics to *host-and-deploy*. This includes moving topics over from the *hosting* folder, which consolidates these subjects. This also includes some UID renaming for consistency and SEO. Additions to the redirect file reflect these changes.
* A UE pass is made. They need a bit more work; however, I did manage to "eat" lot of low-hanging fruit :apple: on this PR:
  * "you"/"your" 2nd person out
  * Future tense dropped
  * Bad spacing addressed
  * Word changes (e.g., contractions, "application" to "app")
  * Touched up a few phrases and sentences
  * Some work done with list formats
  * Metadata updated
* I address #4866 for IIS/MSDeploy, but that issue requires more work for nginx that should go to another author who specializes in that subject matter.
* Additional *web.config* info is added to cover #5055. Specifically, we want to say why it was moved to the content root and where it gets published. However, I elected *NOT* to link the GH issue that discusses moving *web.config* back to the web root (*wwwroot*). That really needs more testing and consultation with the engineering team. I have no confidence that it wouldn't ruin someone's day :boom:.

I've created a [Stage 3 issue](https://github.com/aspnet/Docs/issues/5135) to come back to these topics for additional UE work. An important focus there is the IIS troubleshooting topic. I already know a couple of things I'd like to do to improve it. Let's hold on that for a sec tho. I need to get Antares troubleshooting squared away first (Stage 2), and I'll focus on that this week.

Let's address problems with these updates here; but if something falls into the "nice-to-have" category on review of this PR, I suggest it be entered over on https://github.com/aspnet/Docs/issues/5135 for me to address at Stage 3.

@Rick-Anderson @scottaddie This is ready for :eyes:. cc/ @rachelappel

### CODE REVIEW

File: aspnet/web-forms/overview/deployment/visual-studio-web-deployment/deploying-to-production.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ If you don't already have an Azure account, you can create a free trial account
 ## Create a staging environment
 
 > [!NOTE]
-> Since this tutorial was written, Azure App Service added a new feature to automate many of the processes around having staging and production environments. See [Set up staging environments for web apps in Azure App Service](https://azure.microsoft.com/en-us/documentation/articles/web-sites-staged-publishing/).
+> Since this tutorial was written, Azure App Service added a new feature to automate many of the processes around having staging and production environments. See [Set up staging environments for web apps in Azure App Service](https://azure.microsoft.com/documentation/articles/web-sites-staged-publishing/).
```
Comment (scottaddie) [inline aspnet/web-forms/overview/deployment/visual-studio-web-deployment/deploying-to-production.md]: The words "around having" in the first sentence are a bit vague. Should this say "for setting up" instead?

File: aspnetcore/fundamentals/configuration/index.md [hunk 2]
```diff
@@ -370,7 +370,7 @@ Left: 1988
 
 ## The web.config file
 
-A *web.config* file is required when you host the app in IIS or IIS-Express. *web.config* turns on the AspNetCoreModule in IIS to launch your app. Settings in *web.config* enable the AspNetCoreModule in IIS to launch your app and configure other IIS settings and modules. If you are using Visual Studio and delete *web.config*, Visual Studio will create a new one.
+A *web.config* file is required when hosting the app in IIS or IIS-Express. Settings in *web.config* enable the [ASP.NET Core Module](xref:fundamentals/servers/aspnet-core-module) to launch the app and configure other IIS settings and modules. If the *web.config* file isn't present and the project file references `<Project Sdk="Microsoft.NET.Sdk.Web">`, publishing the project creates a new *web.config* file in the published output (the *publish* folder). For more information, see [Host ASP.NET Core on Windows with IIS](xref:host-and-deploy/iis/index#webconfig).
```
Comment (scottaddie) [inline aspnetcore/fundamentals/configuration/index.md]: - IIS-Express --> IIS Express
- references --> includes
- creates a ~~new~~ *web.config*  file

File: aspnetcore/fundamentals/hosting.md [hunk 3]
```diff
@@ -36,7 +36,7 @@ Create a host using an instance of [WebHostBuilder](/dotnet/api/microsoft.aspnet
...
-* When running behind IIS, enables [IIS integration](xref:publishing/iis). Configures the base path and port the server should listen on when using the [ASP.NET Core Module](xref:fundamentals/servers/aspnet-core-module). The module creates a reverse-proxy between IIS and Kestrel. Also configures the app to [capture startup errors](#capture-startup-errors). For the IIS default options, see [the IIS options section of Host ASP.NET Core on Windows with IIS](xref:publishing/iis#iis-options).
+* When running behind IIS, enables [IIS integration](xref:host-and-deploy/iis/index). Configures the base path and port the server should listen on when using the [ASP.NET Core Module](xref:fundamentals/servers/aspnet-core-module). The module creates a reverse-proxy between IIS and Kestrel. Also configures the app to [capture startup errors](#capture-startup-errors). For the IIS default options, see [the IIS options section of Host ASP.NET Core on Windows with IIS](xref:host-and-deploy/iis/index#iis-options).
```
Comment (scottaddie) [inline aspnetcore/fundamentals/hosting.md]: - should listen on --> listens on
- reverse-proxy --> reverse proxy

File: aspnetcore/fundamentals/hosting.md [hunk 4]
```diff
@@ -55,7 +55,7 @@ Create a host using an instance of [WebHostBuilder](/dotnet/api/microsoft.aspnet
...
+To use IIS as a reverse proxy, call [UseIISIntegration](/aspnet/core/api/microsoft.aspnetcore.hosting.webhostbuilderiisextensions) as part of building the host. `UseIISIntegration` doesn't configure a *server*, like [UseKestrel](/dotnet/api/microsoft.aspnetcore.hosting.webhostbuilderkestrelextensions.usekestrel?view=aspnetcore-1.1) does. `UseIISIntegration` configures the base path and port the server should listen on when using the [ASP.NET Core Module](xref:fundamentals/servers/aspnet-core-module) to create a reverse-proxy between Kestrel and IIS. To use IIS with ASP.NET Core, `UseKestrel` and `UseIISIntegration` must be specified. `UseIISIntegration` only activates when running behind IIS or IIS Express. For more information, see [Introduction to ASP.NET Core Module](xref:fundamentals/servers/aspnet-core-module) and [ASP.NET Core Module configuration reference](xref:host-and-deploy/aspnet-core-module).
```
Comment (scottaddie) [inline aspnetcore/fundamentals/hosting.md]: - should listen on --> listens on
- reverse-proxy --> reverse proxy

File: aspnetcore/fundamentals/servers/aspnet-core-module.md [hunk 5]
```diff
@@ -102,7 +102,7 @@ In ASP.NET Core 1.0, if you call `UseUrls`, call it **before** you call `UseIISI
 
 ### Configure ANCM options in Web.config
 
-Configuration for the ASP.NET Core Module is stored in the *Web.config* file that is located in the application's root folder. Settings in this file point to the startup command and arguments that start your ASP.NET Core app. For sample Web.config code and guidance on configuration options, see [ASP.NET Core Module Configuration Reference](../../hosting/aspnet-core-module.md).
+Configuration for the ASP.NET Core Module is stored in the *Web.config* file that is located in the application's root folder. Settings in this file point to the startup command and arguments that start your ASP.NET Core app. For sample Web.config code and guidance on configuration options, see [ASP.NET Core Module Configuration Reference](xref:host-and-deploy/aspnet-core-module).
```
Comment (scottaddie) [inline aspnetcore/fundamentals/servers/aspnet-core-module.md]: Italicize the Web.config file name

**Comment by guardrex** [comment]
Thanks @scottaddie. I'll hit the *host-and-deploy* topics again at Stage 3 of the work. It's *very* likely that there are more pesky nits to fix in these.

**Comment by Rick-Anderson** [comment]
@guardrex  is this ready to merge?

**Comment by guardrex** [comment]
@Rick-Anderson Yes, ready. :rocket:

**Pull request merged**